gdax/public_client.py

```diff
@@ -8,69 +8,165 @@
 
 
 class PublicClient(object):
-    def __init__(self, api_url="https://api.gdax.com", product_id="BTC-USD"):
-        self.url = api_url.rstrip("/")
+    """GDAX public client API.
+
+    All requests default to the `product_id` specified at object creation if not otherwise specified.
+
+    Attributes:
+        url (str): API endpoint URL
+        product_id (str): Default product used for requests
+
+    """
+
+    def __init__(self, api_url='https://api.gdax.com', product_id='BTC-USD'):
```

Comment:
They are both good options, but I'm leaning towards specifying `product_id` to every function. Being more explicit makes things more readable and easier for new users to pick up. Also; if we are aiming to be as close to the API spec as possible I don't think they have a notion of a default product so this makes sense.